src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -823,6 +815,67 @@ private static IntermediateNode GetAddBinaryOp(IRTranslatorContext context, Texl
                 }
             }
 
+            private static IntermediateNode GetSubBinaryOp(IRTranslatorContext context, TexlBinaryOpNode node, IntermediateNode left, IntermediateNode right, DType leftType, DType rightType)
+            {
+                Contracts.AssertValue(node);
+                Contracts.Assert(node.Op == BinaryOp.Sub);
+
+                switch (leftType.Kind)
+                {
+                    case DKind.Date:
+                        if (rightType == DType.DateTime || rightType == DType.Date)
+                        {
+                            // Date - DateTime => in days
+                            // Date - Date => in days
+                            return new BinaryOpNode(context.GetIRContext(node), BinaryOpKind.DateDifference, left, right);
+                        }
+                        else
+                        {
+                            return new BinaryOpNode(context.GetIRContext(node), BinaryOpKind.SubDateAndDay, left, right);
+                        }
+
+                    case DKind.Time:
+                        if (rightType == DType.Time)
+                        {
+                            // Time - Time => in ms
+                            return new BinaryOpNode(context.GetIRContext(node), BinaryOpKind.SubNumbers, left, right);
+                        }
+                        else
+                        {
+                            // Time + Number
+                            return new BinaryOpNode(context.GetIRContext(node), BinaryOpKind.AddTimeAndMilliseconds, left, right);
```

Comment:
shouldn't one of these be negated, or we use a sub binary op kind?

Reply:
There are new sub binary op kinds.